compliance_checker/imos/imos.py

```diff
@@ -1311,6 +1311,29 @@ def check_quality_control_conventions_for_quality_control_variable(self, dataset
 
         return ret_val
 
+    def check_quality_control_variable_matches_variable(self, dataset):
+        """
+        Check that the name of a quality control variable which is <DATA>_quality_control
+        matches the name of a corresponding data variable <DATA>.
+        """
+        ret_val = []
+        for qc_variable in self._quality_control_variables:
+            qc_variable_root_name = qc_variable.name[:-16] # '_quality_control' is 16 characters long
```

Comment:
The first thing we want to check is that the variable name actually ends in '_quality_control'. e.g.
``` python
result_name = ('qc_var', qc_variable.name, 'name')
reasoning = []
match = re.findall('^(.*)_quality_control$', name)   # returns a list with the root name, if matched
if not match:
    reasoning = ["Name of QC variable should end in '_quality_control'"]
ret_val.append( Result(BaseCheck.HIGH, match==[], result_name, reasoning) )
```